### ISSUE TITLE
App crashing when opening posts in r/Bangkok

### ISSUE THREAD

**Issue by NotGaijin** [Docile-Alligator/Infinity-For-Reddit#1384]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only occurs with a certain link, post, image..., I will include the URL.

### App version

5.4.0

### Where did you get the app from

Google Play

### Android version

10

### Device model

Xiaomi POCOPHONE F1

### First occurred

Today

### Steps to reproduce

These are what I did to confirm the bug
1. Open app
2. Go to r/Bangkok
3. Sort by new
4. Open the top post

### Example post, link, markdown...

https://www.reddit.com/r/Bangkok/comments/11af17u/100_compostable_plastic_straw_and_cup/

### Expected behaviour

Open the post normally

### Current behaviour

The app just crashes

### Logs

```
`` Stacktrace ``

java.lang.IllegalArgumentException: No enum constant ml.docilealligator.infinityforreddit.SortType.Type.BLANK
	at java.lang.Enum.valueOf(Enum.java:257)
	at ml.docilealligator.infinityforreddit.SortType$Type.valueOf(SortType.java:32)
	at ml.docilealligator.infinityforreddit.fragments.ViewPostDetailFragment$14.onFetchSubredditDataSuccess(ViewPostDetailFragment.java:1440)
	at ml.docilealligator.infinityforreddit.subreddit.FetchSubredditData$1$1.onParseSubredditDataSuccess(FetchSubredditData.java:38)
	at ml.docilealligator.infinityforreddit.subreddit.ParseSubredditData$ParseSubredditDataAsyncTask.onPostExecute(ParseSubredditData.java:113)
	at ml.docilealligator.infinityforreddit.subreddit.ParseSubredditData$ParseSubredditDataAsyncTask.onPostExecute(ParseSubredditData.java:79)
	at android.os.AsyncTask.finish(AsyncTask.java:755)
	at android.os.AsyncTask.access$900(AsyncTask.java:192)
	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:772)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:224)
	at android.app.ActivityThread.main(ActivityThread.java:7562)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)


`` END of stacktrace ``

`` Thread info ``

Name: main
ID: 2
State: RUNNABLE
Priority: 5
Thread group name: main
Thread group parent: system
Thread group active count: 56
Thread time: 0 hr 0 min, 1 sec

`` END of thread info ``

`` Device info ``

Report ID: 096fb9cc-2de3-4ea0-968d-41b1c7ee101e
Device ID: 538f20d5ee7b1c0d
Application version: 114
Default launcher: com.mi.android.globallauncher
Timezone name: Japan Standard Time
Timezone ID: Asia/Tokyo
Version release: 10
Version incremental : V12.0.3.0.QEJMIXM
Version SDK: 29
Board: sdm845
Bootloader: unknown
Brand: Xiaomi
CPU ABIS 32: armeabi-v7a, armeabi
CPU ABIS 64: arm64-v8a
Supported ABIS: arm64-v8a, armeabi-v7a, armeabi
Device: beryllium
Display: QKQ1.190828.002 test-keys
Fingerprint: Xiaomi/beryllium/beryllium:10/QKQ1.190828.002/V12.0.3.0.QEJMIXM:user/release-keys
Hardware: qcom
Host: c4-miui-ota-bd48.bj
ID: QKQ1.190828.002
Manufacturer: Xiaomi
Product: beryllium
Build time: 1609053220000
Build time formatted: 27-12-2020 16:13:40
Type: user
Radio: 4.0.c2.6-00335-1227_1222_40a1464,4.0.c2.6-00335-1227_1222_40a1464
Tags: release-keys
User: builder
User IDs: N/A
Is sustained performance mode supported: Yes
Is in power save mode: No
Is in interactive state: Yes
Is ignoring battery optimizations: No
Thermal status: STATUS_NONE
Location power save mode: MODE_NO_CHANGE
Is device idle: No
Battery percentage: 73
Battery remaining time: N/A
Is battery charging: No
Is device rooted: No
CPU Model: Qualcomm Technologies, Inc SDM845
Number of CPU cores: 8
Up time with sleep: 25 hr 1 min, 31 sec
Up time without sleep: 9 hr 20 min, 22 sec

`` END of Device info ``

`` Exit reasons ``

N/A

`` END of exit reasons ``

`` Application info ``

App name: Infinity
Version code: 114
Version name: 5.4.0
Package name: ml.docilealligator.infinityforreddit
Short package name: infinityforreddit
Flavor: N/A
Signatures: XFN80TR5rgSMbsH+WZ5oyxarmIU=
Is debuggable: No
First installed: 24-02-2023 12:20:15
Last updated: 24-02-2023 12:20:15
Requested permissions: android.permission.ACCESS_NETWORK_STATE, android.permission.INTERNET, android.permission.SET_WALLPAPER, android.permission.SET_WALLPAPER_HINTS, android.permission.FOREGROUND_SERVICE, android.permission.WAKE_LOCK, android.permission.POST_NOTIFICATIONS, android.permission.USE_BIOMETRIC, android.permission.USE_FINGERPRINT, android.permission.RECEIVE_BOOT_COMPLETED, ml.docilealligator.infinityforreddit.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION
Default prefs: swipe_up_to_hide_jump_to_next_top_level_comments_button = false, video_autoplay = 2, pinch_to_zoom_video = false, save_sort_type = true, swipe_vertically_to_go_back_from_media = true, disable_swiping_between_tabs = false, show_top_level_comments_first = false, hide_comment_awards = false, click_to_show_media_in_gallery_layout = false, save_nsfw_media_in_different_folder = true, time_format = d MMM yyyy, HH:mm, pull_notification_time = 1677209979602, show_comment_divider = false, video_player_ignore_nav_bar = false, show_author_avatar = false, respect_subreddit_recommended_comment_sort_type = true, default_search_result_tab = 0, title_font_family = Default, font_family = Default, number_of_columns_in_post_feed_landscape = 2, lock_bottom_app_bar = false, comment_toolbar_hidden = false, start_autoplay_visible_area_offset_portrait = 75, separate_folder_for_each_subreddit = true, disable_image_preview = false, user_default_sort_type = NEW, remember_muting_option_in_post_feed = false, volume_keys_navigate_comments = false, hide_subreddit_and_user_prefix = false, long_press_to_hide_toolbar_in_compact_layout = false, post_compact_layout_toolbar_hidden_by_default = false, enable_search_history = false, number_of_columns_in_post_feed_landscape_card_layout_2 = 2, theme = 1, comment_toolbar_hide_on_click = true, video_download_location = content://com.android.externalstorage.documents/tree/644F-E6AA%3ADCIM%2FInfinity%20download, content_font_family = Default, image_download_location = content://com.android.externalstorage.documents/tree/644F-E6AA%3ADCIM%2FInfinity%20download, post_feed_max_resolution = 5000000, subreddit_default_sort_time = ALL, enable_notification = true, hide_the_number_of_votes = false, show_elapsed_time = true, number_of_columns_in_post_feed_portrait_card_layout_2 = 1, show_fewer_toolbar_options_threshold = 5, number_of_columns_in_post_feed_landscape_gallery_layout = 2, number_of_columns_in_post_feed_landscape_unfolded = 2, hide_subreddit_description = false, swipe_between_posts = false, hide_the_number_of_comments = false, data_saving_mode = 0, enable_swipe_action = false, easier_to_watch_in_full_screen = false, content_font_size = Normal, show_only_one_comment_level_indicator = false, swipe_action_threshold = 0.3, immersive_interface_ignore_nav_bar = false, number_of_columns_in_post_feed_portrait_gallery_layout = 2, default_post_layout = 1, mute_autoplaying_videos = true, number_of_columns_in_post_feed_landscape_compact_layout = 2, language = auto, notificaiton_interval = 15, hide_the_number_of_awards = false, start_autoplay_visible_area_offset_landscape = 50, amoled_dark = false, automatically_try_redgifs = true, swipe_to_go_back_from_post_detail = false, lazy_mode_interval = 2.5, hide_text_post_content = false, only_disable_preview_in_video_and_gif_posts = false, hide_post_type = true, autoplay_nsfw_videos = true, main_page_back_button_action = 0, title_font_size = Normal, disable_immersive_interface_in_l
```

**Comment by FibreTTP** [comment]
Also eperiencing this with seemingly unrelated posts, though r/AusLegal seems to always crash for me.

OP's log suggest that there's an issue with comment sort types, so **I turned off _Settings -> Sort Type -> Respect Subreddit Recommended Comment Sort Type_ and I don't crash anymore**.

**Comment by NotGaijin** [comment]
I think this is the fix, it works for me. Thank you.

**Comment by FibreTTP** [comment]
@NotGaijin

I suggest this be re-opened as the crashing is unintended behaviour, and has not been resolved.

**Comment by NotGaijin** [comment]
Reopening the issue as per @FibreTTP 's request

**Comment by Docile-Alligator** [comment]
I think this is fixed in v5.4.1. Could you check that? @NotGaijin

**Comment by NotGaijin** [comment]
Yes, I think it is fixed now. Thank you.
